### PR TITLE
commit for Task 8

### PR DESCRIPTION
**Pull request by StevenJune**
Task 8

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -223,3 +223,54 @@ describe("7. GET /api/articles/:article_id", () => {
...
 });
 
+describe("8. GET /api/articles?topic=:topic", () => {
+  test("status:200, provide topic and responds with array of sorted articles", () => {
+    const topic = "mitch";
+    return request(app)
+      .get(`/api/articles?topic=${topic}`) //${topic}`)
+      .expect(200)
+      .then(({ body }) => {
+        console.log(body, "<<<body");
```
Comment (leekli) [inline __tests__/app.test.js]: Please remove this console log

File: __tests__/app.test.js [hunk 2]
```diff
@@ -223,3 +223,54 @@ describe("7. GET /api/articles/:article_id", () => {
...
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              body: expect.any(String),
```
Comment (leekli) [inline __tests__/app.test.js]: Please re-check the Trello Board instructions - it doesn't want us to return the body key in /api/articles, this needs to be removed in the model and then the test

File: __tests__/app.test.js [hunk 3]
```diff
@@ -223,3 +223,54 @@ describe("7. GET /api/articles/:article_id", () => {
...
+    return request(app)
+      .get(`/api/articles/${article_id}`)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe(
+          `invalid input syntax for type integer: "${article_id}"`
+        );
+      });
+  });
+});
```
Comment (leekli) [inline __tests__/app.test.js:274]: Have a think about some extra tests such as:
* 400 invalid topic
* 404 topic not found
* 200 for an topic that exists but has no articles

**Review by leekli — CHANGES_REQUESTED** [review]
Hi, request for changes:
* Remove console log on line 233 in app.test.js
* Check line 246 in app.test.js - the Trello Board instructions do not want us to return 'body' in the query on /api/articles endpoint
* It may be worth having an expect statement to check how many articles have been returned with the topic of 'mitch', for example if there are 5 articles in the database with a topic of 'mitch' then you would expect a length of 5.
* Have a think about some extra tests such as: 400 invalid topic, 404 topic not found, 200 for an topic that exists but has no articles

Please have a look at these actions, and once complete please submit a new pull request review.

File: models/news.js [hunk 4]
```diff
@@ -68,3 +68,55 @@ exports.selectArticleByIdComment = (article_id) => {
...
+      }
+      return topic;
+    })
+    .then((topic) => {
+      return db.query("select * from articles where topic = $1", [topic]);
+    })
+    .then(({ rows }) => {
+      const result = rows[0];
+      if (!result) {
+        console.log(result, "goto error step 031");
```
Comment (Rayhaan2k9) [inline models/news.js:103]: Please remove the console log

File: models/news.js [hunk 5]
```diff
@@ -68,3 +68,55 @@ exports.selectArticleByIdComment = (article_id) => {
...
+        });
+      }
+      return topic;
+    })
+    .then((topic) => {
+      return db.query(str1, qryparam);
+    })
+    .then(({ rows }) => {
+      const article = rows[0];
+      console.log("setup 04");
```
Comment (Rayhaan2k9) [inline models/news.js:116]: Please remove the console log

**Review by Rayhaan2k9 — APPROVED** [review]
Great work Steven :) Please remove the console logs, make a pull request and then merge it. You don't need to send it to us

**Pull request merged**